### PR TITLE
Acceptance tests update to the new architecture

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

File: spark-bigquery-acceptance-tests/src/main/java/com/google/cloud/spark/bigquery/acceptance/DataprocAcceptanceTestBase.java [hunk 1]
```diff
@@ -196,20 +197,17 @@ public void writeStream() throws Exception {
...
         context.testBaseGcsDir + "/" + testName + "/big_numeric_acceptance_test.zip";
 
-    createZipFile(pyBaseDir, zipFileLocation);
-
-    AcceptanceTestUtils.uploadToGcs(
-        getClass().getResourceAsStream("/acceptance/big_numeric.py"),
-        context.getScriptUri(testName),
-        "text/x-python");
+    // AcceptanceTestUtils.uploadToGcs(
+    //     getClass().getResourceAsStream("/acceptance/big_numeric.py"),
```
Comment (himanshukohli09) [inline spark-bigquery-acceptance-tests/src/main/java/com/google/cloud/spark/bigquery/acceptance/DataprocAcceptanceTestBase.java:206]: why is this commented?

File: bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/ReadRowsHelperTest.java [hunk 2]
```diff
@@ -66,254 +66,254 @@
...
-    int port() {
-      return server.getPort();
-    }
-
-    void stop() throws InterruptedException {
-      server.shutdownNow();
-      server.awaitTermination(10, TimeUnit.SECONDS);
-    }
-  }
+  // // it is not used, we just need the reference
```
Comment (himanshukohli09) [inline bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/ReadRowsHelperTest.java:69]: This is temporarily commented, right?

File: spark-bigquery-dsv1/spark-bigquery-with-dependencies_2.11/src/test/java/com/google/cloud/spark/bigquery/acceptance/Scala211DataprocImage13AcceptanceTest.java [hunk 3]
```diff
@@ -13,22 +13,22 @@
...
 
-  public DataprocImage15AcceptanceTest() {
+  public Scala211DataprocImage13AcceptanceTest() {
     super(context);
   }
 
   @BeforeClass
   public static void setup() throws Exception {
-    context = DataprocAcceptanceTestBase.setup("2.12", "1.5-debian10");
+    context = DataprocAcceptanceTestBase.setup("1.3-debian10");
```
Comment (himanshukohli09) [inline spark-bigquery-dsv1/spark-bigquery-with-dependencies_2.11/src/test/java/com/google/cloud/spark/bigquery/acceptance/Scala211DataprocImage13AcceptanceTest.java:31]: If we are keeping the enum DataprocImage, then we can get this String from the enum.

File: spark-bigquery-dsv1/spark-bigquery-with-dependencies_2.11/src/test/java/com/google/cloud/spark/bigquery/acceptance/Scala211DataprocImage14AcceptanceTest.java [hunk 4]
```diff
@@ -13,22 +13,22 @@
...
 
-  public DataprocImage20AcceptanceTest() {
+  public Scala211DataprocImage14AcceptanceTest() {
     super(context);
   }
 
   @BeforeClass
   public static void setup() throws Exception {
-    context = DataprocAcceptanceTestBase.setup("2.12", "2.0-debian10");
+    context = DataprocAcceptanceTestBase.setup("1.4-debian10");
```
Comment (himanshukohli09) [inline spark-bigquery-dsv1/spark-bigquery-with-dependencies_2.11/src/test/java/com/google/cloud/spark/bigquery/acceptance/Scala211DataprocImage14AcceptanceTest.java:31]: If we are keeping the enum DataprocImage, then we can get this String from the enum

File: spark-bigquery-dsv1/spark-bigquery-with-dependencies_2.12/src/test/java/com/google/cloud/spark/bigquery/acceptance/Scala212DataprocImage15AcceptanceTest.java [hunk 5]
```diff
@@ -13,22 +13,22 @@
...
 
-  public DataprocImage13AcceptanceTest() {
+  public Scala212DataprocImage15AcceptanceTest() {
     super(context);
   }
 
   @BeforeClass
   public static void setup() throws Exception {
-    context = DataprocAcceptanceTestBase.setup("2.11", "1.3-debian10");
+    context = DataprocAcceptanceTestBase.setup("1.5-debian10");
```
Comment (himanshukohli09) [inline spark-bigquery-dsv1/spark-bigquery-with-dependencies_2.12/src/test/java/com/google/cloud/spark/bigquery/acceptance/Scala212DataprocImage15AcceptanceTest.java:31]: If we are keeping the enum DataprocImage, then we can get this String from the enum

File: spark-bigquery-dsv1/spark-bigquery-with-dependencies_2.12/src/test/java/com/google/cloud/spark/bigquery/acceptance/Scala212DataprocImage20AcceptanceTest.java [hunk 6]
```diff
@@ -13,22 +13,22 @@
...
 
-  public DataprocImage14AcceptanceTest() {
+  public Scala212DataprocImage20AcceptanceTest() {
     super(context);
   }
 
   @BeforeClass
   public static void setup() throws Exception {
-    context = DataprocAcceptanceTestBase.setup("2.11", "1.4-debian10");
+    context = DataprocAcceptanceTestBase.setup("2.0-debian10");
```
Comment (himanshukohli09) [inline spark-bigquery-dsv1/spark-bigquery-with-dependencies_2.12/src/test/java/com/google/cloud/spark/bigquery/acceptance/Scala212DataprocImage20AcceptanceTest.java:31]: If we are keeping the enum DataprocImage, then we can get this String from the enum

File: spark-bigquery-dsv2/spark-bigquery-spark24/src/test/java/com/google/cloud/spark/bigquery/acceptance/Spark24DataprocImage14AcceptanceTest.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+  private static AcceptanceTestContext context;
+
+  public Spark24DataprocImage14AcceptanceTest() {
+    super(context, false);
+  }
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    context = DataprocAcceptanceTestBase.setup("1.4-debian10");
```
Comment (himanshukohli09) [inline spark-bigquery-dsv2/spark-bigquery-spark24/src/test/java/com/google/cloud/spark/bigquery/acceptance/Spark24DataprocImage14AcceptanceTest.java:31]: If we are keeping the enum DataprocImage, then we can get this String from the enum

File: spark-bigquery-dsv2/spark-bigquery-spark24/src/test/java/com/google/cloud/spark/bigquery/acceptance/Spark24DataprocImage15AcceptanceTest.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+  private static AcceptanceTestContext context;
+
+  public Spark24DataprocImage15AcceptanceTest() {
+    super(context, false);
+  }
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    context = DataprocAcceptanceTestBase.setup("1.5-debian10");
```
Comment (himanshukohli09) [inline spark-bigquery-dsv2/spark-bigquery-spark24/src/test/java/com/google/cloud/spark/bigquery/acceptance/Spark24DataprocImage15AcceptanceTest.java:31]: If we are keeping the enum DataprocImage, then we can get this String from the enum

File: spark-bigquery-connector-common/src/test/resources/acceptance/big_numeric.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import sys
```
Comment (himanshukohli09) [inline spark-bigquery-connector-common/src/test/resources/acceptance/big_numeric.py:1]: We have two sets of .py files. Can we move them to some common resources directory/folder?

File: spark-bigquery-acceptance-tests/src/main/java/com/google/cloud/spark/bigquery/acceptance/DataprocImage.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
+package com.google.cloud.spark.bigquery.acceptance;
+
+public enum DataprocImage {
```
Comment (himanshukohli09) [inline spark-bigquery-acceptance-tests/src/main/java/com/google/cloud/spark/bigquery/acceptance/DataprocImage.java:3]: This Enum is not used anywhere.

File: bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/ReadRowsHelperTest.java (hunk 2 above)
Comment (davidrabinowitz) [reply bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/ReadRowsHelperTest.java:69]: Yes

File: spark-bigquery-acceptance-tests/src/main/java/com/google/cloud/spark/bigquery/acceptance/DataprocAcceptanceTestBase.java (hunk 1 above)
Comment (davidrabinowitz) [reply spark-bigquery-acceptance-tests/src/main/java/com/google/cloud/spark/bigquery/acceptance/DataprocAcceptanceTestBase.java:206]: The file is uploaded in `createAndRunPythonJob()`

File: spark-bigquery-connector-common/src/test/resources/acceptance/big_numeric.py (hunk 9 above)
Comment (davidrabinowitz) [reply spark-bigquery-connector-common/src/test/resources/acceptance/big_numeric.py]: Done

**Review by himanshukohli09 — APPROVED** [review]

**Pull request merged**